src/main/kotlin/graphics/scenery/volumes/vdi/VDINode.kt

```diff
@@ -0,0 +1,177 @@
+package graphics.scenery.volumes.vdi
+
+import graphics.scenery.RichNode
+import graphics.scenery.ShaderMaterial
+import graphics.scenery.ShaderProperty
+import graphics.scenery.backends.Shaders
+import graphics.scenery.compute.ComputeMetadata
+import graphics.scenery.compute.InvocationType
+import graphics.scenery.textures.Texture
+import graphics.scenery.utils.Image
+import net.imglib2.type.numeric.integer.UnsignedIntType
+import net.imglib2.type.numeric.real.FloatType
+import org.joml.Matrix4f
+import org.joml.Vector3f
+import org.joml.Vector3i
+import org.lwjgl.system.MemoryUtil
+import java.nio.ByteBuffer
+
+class VDINode(windowWidth: Int, windowHeight: Int, val numSupersegments: Int, vdiData: VDIData) : RichNode() {
+
+    @ShaderProperty
+    var ProjectionOriginal = Matrix4f()
+
+    @ShaderProperty
+    var invProjectionOriginal = Matrix4f()
+
+    @ShaderProperty
+    var ViewOriginal = Matrix4f()
+
+    @ShaderProperty
+    var invViewOriginal = Matrix4f()
+
+    @ShaderProperty
+    var ViewOriginal2 = Matrix4f()
+
+    @ShaderProperty
+    var invViewOriginal2 = Matrix4f()
+
+    @ShaderProperty
+    var useSecondBuffer = false
+
+    @ShaderProperty
+    var invModel = Matrix4f()
+
+    @ShaderProperty
+    var volumeDims = Vector3f()
+
+    @ShaderProperty
+    var nw = 0f
+
+    @ShaderProperty
+    var vdiWidth: Int = 0
+
+    @ShaderProperty
+    var vdiHeight: Int = 0
+
+    @ShaderProperty
+    var totalGeneratedSupsegs: Int = 0
+
+    @ShaderProperty
+    var do_subsample = false
+
+    @ShaderProperty
+    var max_samples = 50
+
+    @ShaderProperty
+    var sampling_factor = 0.1f
+
+    @ShaderProperty
+    var downImage = 1f
+
+    @ShaderProperty
+    var skip_empty = true
+
+    @ShaderProperty
+    var stratified_downsampling = false
+
+    @ShaderProperty
+    var printData = true
+
+    private val vulkanProjectionFix =
+        Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, -1.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 0.5f, 0.0f,
+            0.0f, 0.0f, 0.5f, 1.0f)
+
+    fun Matrix4f.applyVulkanCoordinateSystem(): Matrix4f {
```

Comment:
This function should likely be moved into a general utility class, as the renderer is using it as well.

Reply:
Done in 297e8e3f4a2ee330e365d0cbcd5ae750d9ef03b5